cmd/simple-dataplane/simple-dataplane.go

```diff
@@ -148,6 +149,11 @@ func (d DataplaneController) RequestBuildConnect(ctx context.Context, in *simple
 		}, status.Error(codes.Aborted, "failure to interconnect pods")
 	}
 
+	// Add finalizer to both pods, in the event of pod deletion, the controller will be able
+	// to clean up injected dataplane interfaces without any race.
+	_ = finalizerutils.AddPodFinalizer(d.k8s, podName1, podNamespace1)
```

Comment:
Should we be handling errors here and on lien 155?

Reply:
I thought about it before I think not adding finalizer should not prevent pod from starting, after all it is just a before delete protective measure which is nice to have. I do not have a strong opinion here, if you feel that we should fail on this event, I can add it. Please advise.

Reply:
Maybe we should log the error? Without it, I think we should remove the error return value from AddPodFinalizer() and bury it there.

Reply:
done